### PR TITLE
Added filters by location + Added Central option for location

### PR DESCRIPTION
**Pull request by hhjoel**
- Added an additional location "Central" for location
- Added the ability for users to filter by different location
- Use case: users might want to filter restaurant recommendations by locations and so we add this feature here
- Setting multiple filters is done wrongly in GroupServlet

### CODE REVIEW

File: friendly/src/main/webapp/index.html [hunk 1]
```diff
@@ -78,7 +78,7 @@ <h1>Create A New Group</h1>
 
 
         <div id="data">
-            <select id="groups-list"></select>
+            <select id="goups-list"></select>
```
Comment (HeHaowei) [inline friendly/src/main/webapp/index.html]: Why you change it to goups-list?

**Review by HeHaowei — CHANGES_REQUESTED** [review]
I think you accidentally change the "group" to "goup", please check with it. Other parts are quite good, LGTM + 1!

File: friendly/src/main/webapp/index.html (hunk 1 above)
Comment (hhjoel) [reply friendly/src/main/webapp/index.html]: Done!

**Review by HeHaowei — APPROVED** [review]
LGTM + 1!

**Pull request merged**